File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {

        int sum = 0;

        for(int i=0; i<k; i++)
        {
            sum+=nums[i];
        }

        double maxAvg = (double)sum/k;

        for(int i = k;i<nums.size(); i++)
        {
            sum+=nums[i];
            sum-=nums[i-k];

            double avg = (double)sum/k;
            if(avg>maxAvg) maxAvg = avg;
        }

        return maxAvg;
        
    }
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:

        sum = 0
        l = []

        for i in range(0, k):
            sum += nums[i]
        l.append(sum)

        for i in range(k, len(nums)):
            sum+=nums[i] - nums[i-k]
            l.append(sum)

        return max(l)/k

    



        